platform/security/policy/security-policy-context/src/main/java/org/codice/ddf/security/policy/context/impl/PolicyManager.java

```diff
@@ -216,30 +246,137 @@ public synchronized void setPolicies(Map<String, Object> properties) {
+     *
+     * @param path           - Path associated to context
+     * @param contextToRealm - Map of all paths and their context realms
+     * @return context - Realm associated to path
+     */
+    public String getContextRealm(String path, Map<String, String> contextToRealm) {
+        String entry = contextToRealm.get(path);
+        if (entry != null) {
+            return entry;
+        } else {
+
+            //continue splitting by last "/"  down path until value found
+            int idx = path.lastIndexOf("/");
+            if (idx <= 0) {
+                idx++;
+            }
+            String pathFragment = path.substring(0, idx);
+            if (StringUtils.isNotEmpty(pathFragment)) {
+                return getContextRealm(pathFragment, contextToRealm);
+            } else {
+                return DEFAULT_REALM_CONTEXT_VALUE;
             }
         }
+    }
 
-        currentPolicies = Collections.unmodifiableCollection(new ArrayList<>(policyStore.values()));
+    /**
+     * Gets the authorization types of the given path. If authorization types of given path do not exist it rolls back until a parent path exists with such authorization types is found
+     * If no parent path exists, the authorization types defaults to an empty list
+     *
+     * @param path               - Path associated with context
+     * @param contextToAuthTypes - Map of all paths and their context authorization types
+     * @return List of authorization types
+     */
+    public List<String> getContextAuthTypes(String path,
+            Map<String, List<String>> contextToAuthTypes) {
+        List<String> entry = contextToAuthTypes.get(path);
+        if (entry != null) {
+            return entry;
+        } else {
 
-        LOGGER.debug("Policy store initialized, now contains {} entries", policyStore.size());
+            //continue splitting by last "/"  down path until value found, if last "/" is hit return default authentication type
+            int idx = path.lastIndexOf("/");
+            if (idx <= 0) {
+                idx++;
+            }
+            String pathFragment = path.substring(0, idx);
+            if (StringUtils.isNotEmpty(pathFragment)) {
+                return getContextAuthTypes(pathFragment, contextToAuthTypes);
+            } else {
+                return new ArrayList<String>();
+            }
+        }
+    }
+
+    /**
+     * Gets the required attributes of the context associated to the given path. If required attributes of given path do not exist it rolls back until a parent context path exists with such required attributes is found
+     * If no parent path exists, the context defaults to empty list
+     *
+     * @param -                 Path associated to context
+     * @param contextToReqAttrs - Map of all paths to contexts and their associated required attributes
+     * @return List of required attributes
+     */
+    public List<ContextAttributeMapping> getContextReqAttrs(String path,
```

Comment:
These three methods all do the same processing and then operate on the three different collections. Begs for a closure (soon, but not yet) or a strategy. Actually, you could do the pre-Java 8 version of a closure with a guava Function or a Callable.